wasp/drivers/battery.py

```diff
@@ -50,26 +51,34 @@ def voltage_mv(self):
 
         Assumes a 50/50 voltage divider and a 3.3v power supply
 
+        The last values is kept in a cache and only the minium cached value is
+        shown to the user, this is to avoid the battery level
+        going up and down because of the lack of precision of the mv.
+        Note that this will underestimate battery level.
+
         :returns: Battery voltage, in millivolts.
         """
         raw = self._battery.read_u16()
-        return (2 * 3300 * raw) // 65535
+        mv = (2 * 3300 * raw) // 65535
+
+        if mv not in self._cache:
+            self._cache.add(mv)
+            while len(self._cache) > 2:
+                self._cache.remove(max(self._cache))
+        return min(self._cache)
```

Comment:
won't this always keep the minimum voltage?
for example
|loop|mv|cache|min|
|-|-|-|-|
|0|1|{1}|1|
|1|2|{1,2}|1|
|2|5|{1,2}|1|
|...|...|{1,2}|1|